vulture/config.py

```diff
@@ -0,0 +1,212 @@
+"""
+This module handles retrieval of configuration values from either the
+command-line arguments or the pyproject.toml file.
+"""
+import argparse
+import sys
+from os.path import abspath, exists
+
+import toml
+
+from .version import __version__
+
+#: Possible configuration options and their respective defaults
+DEFAULTS = {
+    "min_confidence": 0,
+    "paths": [],
+    "exclude": [],
+    "ignore_decorators": [],
+    "ignore_names": [],
+    "make_whitelist": False,
+    "sort_by_size": False,
+    "verbose": False,
+}
+
+#: sentinel value to distinguish between "False" and "no default given"
+MISSING = object()
+
+
+def _check_input_config(data):
+    """
+    Checks the types of the values in *data* against the expected types of
+    config-values. If a value is of the wrong type it will raise a SystemExit.
+    """
+    for key, value in data.items():
+        if key not in DEFAULTS:
+            print(f"Unknown configuration key: {key}", file=sys.stderr)
+            sys.exit("Invalid config")
+        if value is not MISSING and not type(value) is type(  # noqa: E721
```

Comment:
I think it's a pity that there's no better way to solve this, but it's ok to leave the noqa there. Although I'd be interested to see an example of where isinstance screws things up here.

Reply:
The issue is with `int` and `bool`. Because `isinstance(False, int)` returns `True`.

Reply:
Ah, thanks!